pyrsistent/_precord.py

```diff
@@ -123,7 +126,10 @@ def set(self, key, original_value):
         if field:
             if self._factory_fields is None or field in self._factory_fields:
                 try:
-                    value = field.factory(original_value)
+                    if is_type_cls(PRecord, field.type):
```

Comment:
I'm not sure that this is a safe assumption. What if there is a custom `factory` method for the field? Then you would pass an extra parameter, `ignore_extra`, that the factory did not expect. You would probably have to check that the factory is actually the types `create` method as well.

I would prefer to guard all this processing behind an if on ´self._ignore_extra´ to avoid affecting the runtime performance too much with digging around in class hierarchies etc. It would also make it clear to the reader of the code that this code is only needed for propagating that parameter.

Reply:
Done